flink-python/pyflink/batch/execution_environment.py

```diff
@@ -0,0 +1,174 @@
+# limitations under the License.
+################################################################################
+from pyflink.common.execution_config import ExecutionConfig
+from pyflink.common.restart_strategy import RestartStrategies
+from pyflink.java_gateway import get_gateway
+from pyflink.util.utils import load_java_class
+
+
+class ExecutionEnvironment(object):
+    """
+    The ExecutionEnvironment is the context in which a program is executed.
+
+    The environment provides methods to control the job execution (such as setting the parallelism)
+    and to interact with the outside world (data access).
+    """
+
+    def __init__(self, j_execution_environment):
+        self._j_execution_environment = j_execution_environment
+
+    def get_parallelism(self):
+        """
+        Gets the parallelism with which operation are executed by default.
+
+        :return: The parallelism.
+        """
+        return self._j_execution_environment.getParallelism()
+
+    def set_parallelism(self, parallelism):
+        """
+        Sets the parallelism for operations executed through this environment.
+        Setting a parallelism of x here will cause all operators to run with
+        x parallel instances.
+
+        :param parallelism: The parallelism.
+        """
+        self._j_execution_environment.setParallelism(parallelism)
+
+    def get_default_local_parallelism(self):
+        """
+        Gets the default parallelism that will be used for the local execution environment.
+
+        :return: The parallelism.
+        """
+        return self._j_execution_environment.getDefaultLocalParallelism()
+
+    def set_default_local_parallelism(self, parallelism):
+        """
+        Sets the default parallelism that will be used for the local execution environment.
+
+        :param parallelism: The parallelism.
+        """
+        self._j_execution_environment.setDefaultLocalParallelism(parallelism)
+
+    def get_config(self):
+        """
+        Gets the config object that defines execution parameters.
+
+        :return: The environment's execution configuration.
+        """
+        return ExecutionConfig(self._j_execution_environment.getConfig())
+
+    def set_restart_strategy(self, restart_strategy_configuration):
+        """
+        Sets the restart strategy configuration. The configuration specifies which restart strategy
+        will be used for the execution graph in case of a restart.
+
+        :param restart_strategy_configuration: Restart strategy configuration to be set.
+        """
+        self._j_execution_environment.setRestartStrategy(
+            restart_strategy_configuration._j_restart_strategy_configuration)
+
+    def get_restart_strategy(self):
+        """
+        Returns the specified restart strategy configuration.
+
+        :return: The restart strategy configuration to be used.
+        """
+        return RestartStrategies._from_j_restart_strategy(
```

Comment:
Sometimes the conversion from java type to python type is a classmethod, sometimes it's a utility method. We should keep it consistent. What about declare all these methods static?